### PR TITLE
[FIX] purchase: make fiscal position computed

### PR DESCRIPTION
**Pull request by guva-odoo**
Making the fiscal position computed allows it to be
correctly set when a PO is created automatically,
for example from a sale order with inter-company rules.

opw-4247948

### CODE REVIEW

**Comment by guva-odoo** [comment]
enterprise PR: https://github.com/odoo/enterprise/pull/73392

**Comment by guva-odoo** [comment]
hello @svs-odoo could you take a little time to review this one please?

File: addons/purchase/models/purchase.py [hunk 1]
```diff
@@ -189,6 +194,23 @@ def _compute_date_planned(self):
...
+        for order in self:
+            if not order.partner_id:
+                order.fiscal_position_id = False
+                continue
+            key = (order.company_id.id, order.partner_id.id)
+            if key not in cache:
+                cache[key] = self.env['account.fiscal.position'].with_company(
+                    order.company_id
+                )._get_fiscal_position(order.partner_id)
+            order.fiscal_position_id = cache[key]
```
Comment (ticodoo) [inline addons/purchase/models/purchase.py:212]: I think it's a bit risky to change this in stable. There are already a lot of spots where the `fiscal_position_id` is being pre-calculated and passed as the create values for a PO (these are only a couple of them, there are more):

https://github.com/odoo/odoo/blob/9d4f8b2fda5c75438ad30a08b4f7d95eac4f2f23/addons/sale_purchase/models/sale_order_line.py#L129

https://github.com/odoo/odoo/blob/9d4f8b2fda5c75438ad30a08b4f7d95eac4f2f23/addons/purchase_stock/models/stock_rule.py#L284

Also, at this point there may also be lots of customizations that depend on the current onchange behavior/pre-calc create values.

Final also, `order.partner_id` is a mandatory field, so this compute is always going to go through every partner/company key combo.

**Review by ticodoo — COMMENTED** [review]
IMO, this change would be ok in master with lots of cleaning up of all the places that are currently calling `_get_fiscal_position()`, but it's too risky for stable. Could break lots of stuff for fixing the very specific use case of inter-company rules. It would be better to add a fix that only applies to the specific use case + add a test for it

File: addons/purchase/models/purchase.py (hunk 1 above)
Comment (guva-odoo) [reply addons/purchase/models/purchase.py:212]: Never mind, I just took another look, and we don't even need that, I don't know why I did that in the first place. Thanks!

**Comment by guva-odoo** [comment]
Not needed

**Pull request closed without merging**